typescript/infra/src/core/check.ts

```diff
@@ -193,58 +180,55 @@ export class AbacusCoreChecker extends AbacusAppChecker<
     }
   }
 
-  async checkInboxes(domain: types.Domain): Promise<void> {
+  async checkInboxes(network: Networks): Promise<void> {
+    const coreContracts = this.app.getContracts(network);
+
     // Check that all inboxes on this domain are pointed to the right validator
     // manager.
-    const contracts = this.app.mustGetContracts(domain);
-
-    await Promise.all(
-      this.app.remoteDomainNumbers(domain).map(async (remote) => {
-        const expectedValidatorManager = this.app.mustGetInboxValidatorManager(
-          remote,
-          domain,
-        );
-        expect(
-          await this.app.mustGetInbox(remote, domain).validatorManager(),
-        ).to.equal(expectedValidatorManager.address);
+    await utils.promiseObjAll(
+      utils.objMap(coreContracts.inboxes, async (_, inbox) => {
+        const expected = inbox.validatorManager.address;
+        const actual = await inbox.inbox.validatorManager();
+        expect(actual).to.equal(expected);
       }),
     );
 
     // Check that all inboxes on this domain share the same implementation and
     // UpgradeBeacon.
-    const inboxes = Object.values(contracts.addresses.inboxes);
+    const coreAddresses = this.app.getAddresses(network);
+    const inboxes: MailboxAddresses[] = Object.values(coreAddresses.inboxes);
     const implementations = inboxes.map((r) => r.implementation);
     const identical = (a: any, b: any) => (a === b ? a : false);
     const upgradeBeacons = inboxes.map((r) => r.beacon);
     expect(implementations.reduce(identical)).to.not.be.false;
     expect(upgradeBeacons.reduce(identical)).to.not.be.false;
   }
 
-  async checkAbacusConnectionManager(domain: types.Domain): Promise<void> {
-    const contracts = this.app.mustGetContracts(domain);
-    for (const remote of this.app.remoteDomainNumbers(domain)) {
-      // inbox is enrolled in abacusConnectionManager
-      const enrolledInboxes =
-        await contracts.abacusConnectionManager.getInboxes(remote);
-      expect(enrolledInboxes).to.eql([
-        this.app.mustGetInbox(remote, domain).address,
-      ]);
-    }
+  async checkAbacusConnectionManager(network: Networks): Promise<void> {
+    const coreContracts = this.app.getContracts(network);
+    await utils.promiseObjAll(
```

Comment:
Do you generally prefer to import `utils` and then access properties on it or destructure in the import? I have a slight preference for the latter